src/viewProviders/baseModels/base.ts

```diff
@@ -144,10 +144,18 @@ export abstract class BaseViewProvider<T extends BaseViewProviderData>
         // special URI scheme to decorate the tree item with a dot to the right of the label,
         // and color the label, description, and decoration so it stands out in the tree view
         treeItem.resourceUri = Uri.parse(`${SEARCH_DECORATION_URI_SCHEME}:/${element.id}`);
+      } else {
+        // clear any previous search decoration URI
+        // NOTE: this will need to be adjusted for https://github.com/confluentinc/vscode/issues/1777
+        treeItem.resourceUri = undefined;
```

Comment:
Special attention here for reviewers: this change was needed to prevent search result decorations from not being cleared when the view's search string was changed since we're reusing the same container item instances (necessary for per-item updates) and not recreating them on refresh.